### PR TITLE
Make domain suffix configurable

### PR DESCRIPTION
**Pull request by kyessenov**
Allow a different DNS domain suffix than "cluster.local"

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -207,6 +207,8 @@ func init() {
 		"Select a namespace for the controller loop. If not set, uses ${POD_NAMESPACE} environment variable")
 	rootCmd.PersistentFlags().DurationVar(&flags.controllerOptions.ResyncPeriod, "resync", time.Second,
 		"Controller resync interval")
+	rootCmd.PersistentFlags().StringVar(&flags.controllerOptions.DomainSuffix, "domainSuffix", "cluster.local",
```
Comment (lookuptable) [inline cmd/manager/main.go]: The flag used by kubeadm to specify this is called `--service-dns-domain`. Shall we just use the same thing?

File: platform/kube/controller.go [hunk 2]
```diff
@@ -606,7 +607,7 @@ func (c *Controller) GetIstioServiceAccounts(hostname string, ports []string) []
 		}
 
 		pod, _ := item.(*v1.Pod)
-		sa := generateServiceAccountID(pod.Spec.ServiceAccountName, pod.GetNamespace(), localDomain)
+		sa := generateServiceAccountID(pod.Spec.ServiceAccountName, pod.GetNamespace(), c.domainSuffix)
```
Comment (lookuptable) [inline platform/kube/controller.go]: CA needs to support similar flag.

Do you think service and service account should always have the same domain?

File: cmd/manager/main.go (hunk 1 above)
Comment (kyessenov) [inline cmd/manager/main.go]: Localkube calls it `dns-domain`, kubelet calls it `cluster-domain`. Seems like a lot of options :)

File: platform/kube/controller.go (hunk 2 above)
Comment (kyessenov) [inline platform/kube/controller.go]: You should be using service account domain name, not service domain name. Not sure if they are the same thing.

**Pull request merged**